iis/datadog_checks/iis/iis.py

```diff
@@ -42,35 +46,31 @@
 class IIS(PDHBaseCheck):
     SERVICE_CHECK = "iis.site_up"
 
-    def __init__(self, name, init_config, agentConfig, instances):
-        super(IIS, self).__init__(name, init_config, agentConfig, instances=instances, counter_list=DEFAULT_COUNTERS)
+    def __init__(self, name, init_config, instances):
+        super(IIS, self).__init__(name, init_config, instances=instances, counter_list=DEFAULT_COUNTERS)
+        self.sites = self.instance.get('sites') or []
 
-    def get_iishost(self, instance):
-        inst_host = instance.get("host")
+    def get_iishost(self):
+        inst_host = self.instance.get("host")
         if inst_host in [".", "localhost", "127.0.0.1", None]:
             # Use agent's hostname if connecting to local machine.
-            iis_host = self.hostname
+            iis_host = datadog_agent.get_hostname()
         else:
             iis_host = inst_host
         return "iis_host:{}".format(self.normalize_tag(iis_host))
 
-    def check(self, instance):
-        # Duplicate refresh_counters from PDHBaseCheck
-        instance_hash = hash_mutable(instance)
-        refresh_counters = is_affirmative(instance.get('refresh_counters', True))
-
-        if refresh_counters:
+    def check(self, _):
+        if self.refresh_counters:
             for counter, values in list(iteritems(self._missing_counters)):
-                self._make_counters(instance_hash, ([counter], values))
+                self._make_counters(self.instance_hash, ([counter], values))
 
-        sites = instance.get('sites', []) or []
-        expected_sites = set(sites)
+        expected_sites = set(self.sites)
         # _Total should always be in the list of expected sites; we always
         # report _Total
         expected_sites.add(TOTAL_SITE)
-
         self.log.debug("Expected sites is %s", expected_sites)
-        for inst_name, dd_name, metric_func, counter in self._metrics[instance_hash]:
+
+        for inst_name, dd_name, metric_func, counter in self._metrics[self.instance_hash]:
```

Comment:
We should have been able to completely get rid of instance_hash, not sure why it's still using it. I guess for compatibility reasons?

Reply:
Yes, I can only get rid if it if update base class and children classes on same PR